src/legacy/core_plugins/kibana/public/discover/saved_searches/saved_searches_register.ts

```diff
@@ -31,13 +31,13 @@ savedObjectManagementRegistry.register({
   service: 'savedSearches',
   title: 'searches',
 });
+const services = {
+  savedObjectsClient: npStart.core.savedObjects.client,
+  indexPatterns: npStart.plugins.data.indexPatterns,
+  chrome: npStart.core.chrome,
+  overlays: npStart.core.overlays,
+};
+const savedSearches = createSavedSearchesService(services);
+
 const module = uiModules.get('discover/saved_searches');
-module.service('savedSearches', () => {
-  const services = {
-    savedObjectsClient: npStart.core.savedObjects.client,
-    indexPatterns: npStart.plugins.data.indexPatterns,
-    chrome: npStart.core.chrome,
-    overlays: npStart.core.overlays,
-  };
-  return createSavedSearchesService(services);
-});
+module.service('savedSearches', () => savedSearches);
```

Comment:
We could use the same pattern as suggested above for dashboard here (only creating the service once and exposing through the contract, but that way is also fine because it's just created twice, once for "inside" the app and once for the global registry. That approach would also be OK for me for dashboard.

Reply:
♩ ♪ ♫ ♬ ♭ ♮ ♯ 🎼 🎵 🎶  I did it discover way ♩ ♪ ♫ ♬ ♭ ♮ ♯ 🎼 🎵 🎶